### PR TITLE
set max_connection from code (when less than 4 is needed)

### PR DESCRIPTION
**Pull request by Stefano-Orsolini**
not very common but some might require to set the maximum number of clients, from code, to less than 4.
in my case i must allow only one client at a time.

### CODE REVIEW

File: libraries/ESP8266WiFi/src/ESP8266WiFiAP.cpp [hunk 1]
```diff
@@ -66,6 +66,9 @@ static bool softap_config_equal(const softap_config& lhs, const softap_config& r
     if(lhs.ssid_hidden != rhs.ssid_hidden) {
         return false;
     }
+    if(lhs.max_connection != rhs.max_connection) {
+    return false;
```
Comment (igrr) [inline libraries/ESP8266WiFi/src/ESP8266WiFiAP.cpp]: Please check indentation of this line

File: libraries/ESP8266WiFi/src/ESP8266WiFiAP.cpp [hunk 2]
```diff
@@ -81,7 +84,7 @@ static bool softap_config_equal(const softap_config& lhs, const softap_config& r
  * @param channel       WiFi channel number, 1 - 13.
  * @param ssid_hidden   Network cloaking (0 = broadcast SSID, 1 = hide SSID)
  */
```
Comment (igrr) [inline libraries/ESP8266WiFi/src/ESP8266WiFiAP.cpp]: function comment block needs to be updated to include new argument

Comment (Stefano-Orsolini) [inline libraries/ESP8266WiFi/src/ESP8266WiFiAP.cpp]: included

File: libraries/ESP8266WiFi/src/ESP8266WiFiAP.cpp (hunk 1 above)
Comment (Stefano-Orsolini) [inline libraries/ESP8266WiFi/src/ESP8266WiFiAP.cpp]: corrected

**Pull request merged**